src/lazy/system_reader.rs

```diff
@@ -333,4 +430,262 @@ mod tests {
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+                $ion_symbol_table::{
+                    imports: [ { name:"shared_table", version: 1 } ],
+                    symbols: [ "local_symbol" ]
+                }
+                $10 // "foo"
+                $11 // "local_symbol"
+            "#,
+            map_catalog,
+        );
+        // We step over the LST...
+        let _symtab = reader.next_item()?.expect_symbol_table()?;
+        // ...but expect all of the symbols we encounter after it to be in the symbol table,
+        // indicating that the SystemReader processed the LST even though we skipped it with `next()`
+        assert_eq!(
+            reader
+                .next_item()?
+                .expect_value()?
+                .read()?
+                .expect_symbol()?,
+            "foo"
+        );
+        assert_eq!(
+            reader
+                .next_item()?
+                .expect_value()?
+                .read()?
+                .expect_symbol()?,
+            "local_symbol"
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn multiple_shared_symbol_table_imports() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        map_catalog.insert_table(SharedSymbolTable::new("shared_table_1", 1, ["foo"])?);
+        map_catalog.insert_table(SharedSymbolTable::new("shared_table_2", 1, ["bar"])?);
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+                // This symbol table will be overwritten by the following one
+                $ion_symbol_table::{
+                    symbols: [ "potato salad" ]
+                }
+                // This LST does not import `$ion_symbol_table`, so it resets rather than appending
+                $ion_symbol_table::{
+                    imports: [ { name:"shared_table_1", version: 1 }, { name:"shared_table_2", version: 1 } ],
+                    symbols: [ "local_symbol" ]
+                }
+                $10 // "foo"
+                $11 // "bar"
+                $12 // "local_symbol"
+          "#,
+            map_catalog,
+        );
+        // We step over the first LST, processing it but not yet applying it.
+        let _symtab = reader.next_item()?.expect_symbol_table()?;
+        // There are 10 symbols in the symbol table, $0 to $9 inclusive. $10 is not defined yet
+        // because we're still parked on the symtab.
+        assert_eq!(reader.symbol_table().len(), 10);
+        // The reader has analyzed the symtab struct and identified what symbols will be added when
+        // it advances beyond it.
+        assert_eq!(
+            reader.pending_symtab_changes().local_symbols()[0].text(),
+            Some("potato salad")
+        );
```

Comment:
🗺️ This is a demonstration of the new tooling access to pending symbol table changes.